pyfesom2/datasets.py

```diff
@@ -0,0 +1,237 @@
+        "Dataset URL": "https://swiftbrowser.dkrz.de/public/dkrz_02942825-0cab-44f3-ad37-80fd5d2e37e3/FESOM2_data/A01"}
+}
+
+frontier_datasets = {
+    "ROSSBY4.2"        : {
+        "path_url"   : "https://swift.dkrz.de/v1/dkrz_035d8f6ff058403bb42f8302e6badfbc/pyfesom2/frontier/rossby42",
+        "Dataset URL": "https://swiftbrowser.dkrz.de/public/dkrz_035d8f6ff058403bb42f8302e6badfbc/pyfesom2/frontier/rossby42"},
+    "ROSSBY4.2_level"  : {
+        "path_url"   : "https://swift.dkrz.de/v1/dkrz_035d8f6ff058403bb42f8302e6badfbc/pyfesom2/frontier/rossby42_level_aceess",
+        "Dataset URL": "https://swiftbrowser.dkrz.de/public/dkrz_035d8f6ff058403bb42f8302e6badfbc/pyfesom2/frontier/rossby42_level_aceess"},
+    "ROSSBY4.2_spatial": {
+        "path_url"   : "https://swift.dkrz.de/v1/dkrz_035d8f6ff058403bb42f8302e6badfbc/pyfesom2/frontier/rossby42_spatial_aceess",
+        "Dataset URL": "https://swiftbrowser.dkrz.de/public/dkrz_035d8f6ff058403bb42f8302e6badfbc/pyfesom2/frontier/rossby42_spatial_aceess"}}
+
+all_datasets = {
+    "LCORE"  : {
+        "path_url"   : "https://swift.dkrz.de/v1/dkrz_02942825-0cab-44f3-ad37-80fd5d2e37e3/FESOM2_data/LCORE",
+        "Dataset URL": "https://swiftbrowser.dkrz.de/public/dkrz_02942825-0cab-44f3-ad37-80fd5d2e37e3/FESOM2_data/LCORE",
+        "var_list"   : ["temp", "salt", "a_ice", "m_ice", "ssh", "sst"]},
+    "pi-grid": {
+        'path_url'   : "https://swift.dkrz.de/v1/dkrz_035d8f6ff058403bb42f8302e6badfbc/pyfesom2/tutorial/pi-grid",
+        "var_list"   : ['a_ice', 'm_ice', 'temp', 'u', 'v', 'w', 'mesh'],
+        "Dataset URL": "https://swiftbrowser.dkrz.de/public/dkrz_035d8f6ff058403bb42f8302e6badfbc/pyfesom2/tutorial/pi-grid"},
+    **cmip6_grids,
+    **frontier_datasets
+}
+
+
+class RemoteZarrDataset:
+    """Fetches a remote Zarr dataset.
+
+    Dataset is only loaded on .load() for dataset containg more variables
+    """
+
+    def __init__(self, path_url, var_list=None, consolidated=True, **kwargs):
+        self.path_url = path_url  # can also be local path remove fsspec in that case
+        self.var_list = var_list
+        self.is_consolidated = consolidated
+        self.dset_attrs = kwargs
+        self._ds = None
+
+    @property
+    def merged_dataset(self):
+        import fsspec  # delay import
+        if self._ds is None:
+            if self.var_list is not None:
+                urls = [self.path_url + "/" + var for var in self.var_list]
+                dataset_list = [xr.open_zarr(fsspec.get_mapper(url), consolidated=self.is_consolidated) for url in urls]
+                self._ds = xr.merge(dataset_list)
+            else:
+                self._ds = xr.open_zarr(self.path_url, consolidated=True)
+
+            self._ds.attrs.update(self.dset_attrs)
+        return self._ds
+
+    def load(self):
+        return self.merged_dataset
+
+
+lcore = RemoteZarrDataset(**all_datasets['LCORE'])
```

Comment:
Please rename it to just `core`. Also in other places through the code :)

Reply:
Okay, will also add atleast minimal doc strings for couple of non doc'd methods before merge.

Reply:
I am done, if you mark it resolved I will merge or you can :)